src/app/clients/add/AddClientForm.tsx

```diff
@@ -176,20 +177,27 @@ const AddClientForm: React.FC = () => {
 
         const inputError = checkErrorOnSubmit(formErrors, setFormErrors);
         if (inputError) {
-            setSubmitError(Error.submit);
+            setSubmitError(Errors.submit);
             setSubmitDisabled(false);
             return;
         }
-        const familyID = await insertClient(clientRecord);
-        const insertSuccess = await insertFamily([...fields.adults, ...fields.children], familyID);
-        if (!insertSuccess[0]) {
-            setSubmitError(Error.external);
-            setSubmitErrorMessage(insertSuccess[1]);
-            setSubmitDisabled(false);
-            return;
+
+        try {
+            const ids = await insertClient(clientRecord);
+            try {
+                await insertFamily([...fields.adults, ...fields.children], ids.family_id);
```

Comment:
shame supabase doesn't support 'transactions', I can get something to work via postgres but I think this is fine for now